developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,421 @@
+---------------------
+**Usage:**
+``domain_types type``
+
+**Example:**
+``domain_types DP``
+
+Following domain types are available :
+- *LL* - low latency domain is intended to be able transfer data through DSP
+subsystem within 2ms. Processing modules used in that domain must be able to
+work on limited number of samples. To meet 2ms pipeline latency on stream
+sampled at 48 kHz stream needs to be processed in 1ms periods i.e. using maximum
+48 sample blocks.
+- *DP* - Data processing domain is intended host all processing that does not
+fit in definition of low latency domain. It is important that this domain is
+still real time and latency sensitive.
+
+Module type
+-----------
+**Usage:**
+``type class_name``
+
+**Example:**
+``type AudClassModule``
+
+*class_name* refers to module class type supported by firmware. Available
+modules classes are placed in:
+``<repo_dir>/FW/intel_common/module_binmaps/dsp_fw_common.binmap``
+in ``# type <mod_func_type>`` section.
+
+.. code-block:: bash
+
+  enum module_type {
+        basefw          = 0,
+        mixin           = 1,
+        mixout          = 2,
+        copier          = 3,
+        peakvol         = 4,
+        updwmix         = 5,
+        mux             = 6,
+        src             = 7,
+        wov             = 8,
+        fx              = 9,
+        aec             = 10,
+        kpb             = 11,
+        micselect       = 12,
+        fxf             = 13,
+        audclass        = 14,
+        fakecopier      = 15,
+        iodriver        = 16,
+        whm             = 17,
+        gdbstub         = 18,
+        sensing         = 19,
+        max             = 20,
+        invalid         = emax
+   };
+
+auto_start
+----------
+**Usage:**
+``auto_start true or false`` Indicates whether a instance of the module should be created at the base fw startup
+
+**Example:**
+``auto_start false``
+
+init_config
+-----------
+**Usage:**
+``init_config true or false`` 0 (only basic config in payload) or 1 (basic config with extension that contains pin format).
+
+**Example:**
+``init_config 1``
+
+
+Module stack size
+-----------------
+**Usage:**
+``stack_size size``
+
+**Example:**
+``stack_size 1000``
+
+*stack_size* is size of stack that module instance requires for its task. It refer only to DP scheduler domain.
+
+Scheduling capabilities
+-----------------------
+**Usage:**
+``sched_caps scheduling_period multiples_supported``
+
+**Example:**
+``sched_caps 1 all``
+
+- *scheduling_period* - scheduling period should be given in samples per channel
+  (i.e. 1 sample = 1 sample per channel) as a hexadecimal value.
+- *multiples_supported* - indicates the supported period multiples. Available
+  values are placed in:
+
+``<repo_dir>/FW/intel_common/module_binmaps/dsp_fw_common.binmap``
+in ``# multiples_supported <list_of_supported_multiples>`` section.
+
+Module config
+-------------
+**Usage:**
+``mod_cfg PAR_0 PAR_1 PAR_2 PAR_3 IS_BYTES CPS IBS OBS MOD_FLAGS CPC OBLS``
```

Comment:
this is a bit long, 11 args is a bit hard to read

Reply:
what's your suggestion ? It defines array of config data.

Reply:
The indicies are named no? Can they not be broken out?

Reply:
sorry, I don't get your idea.  Can you give an example ? Thanks!

Reply:
Based on the description even though it is a byte array, the bytes themselves are all mapped directly to a value. Just curious why we cannot treat this as a packed struct and give everything a name and/or possible sub structs to make access and identification easier.

Reply:
In my opinion, we can define struct in C language code, but this is a config file and these data blob are generated by FW tools, so like topology setting, we just include byte array.